scipy/signal/tests/test_signaltools.py

```diff
@@ -267,6 +267,19 @@ def test_fillvalue(self):
                    [32, 46, 67, 62, 48]])
         assert_array_equal(c, d)
 
+    def test_fillvalue_direct_cast(self):
+        # These is an example that used to work, but it seems odd enough
+        # to disallow directly?
```

Comment:
OK, so actually it is not as odd as I first thought, since these fail:
```
np.array(complex(123. + 1j), dtype=np.float64)
np.array(complex(123. + 1j), dtype=np.intp)
```
So giving a complex fill value (python scalar, not numpy array) would be give error after this if the convolution is not on complex arrays. Which frankly seems right.